### PR TITLE
Added e2e test for batch API group

### PR DESCRIPTION
**Pull request by erictune**

### CODE REVIEW

**Comment by fejta** [comment]
@erictune please assign back to me once tests are passing.

**Comment by erictune** [comment]
@fejta ready for you

File: test/e2e/batch_v1_jobs.go [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+
+const (
+	// How long to wait for a job to finish.
+	v1JobTimeout = 15 * time.Minute
+
+	// Job selector name
+	v1JobSelectorKey = "job-name"
+)
+
+var _ = Describe("V1Job", func() {
```
Comment (fejta) [inline test/e2e/batch_v1_jobs.go]: Should the test name contain batch?

File: test/e2e/batch_v1_jobs.go [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+})
+
+// newTestV1Job returns a job which does one of several testing behaviors.
+func newTestV1Job(behavior, name string, rPol api.RestartPolicy, parallelism, completions int) *extensions.Job {
+	job := &extensions.Job{
+		ObjectMeta: api.ObjectMeta{
+			Name: name,
+		},
+		Spec: extensions.JobSpec{
+			Parallelism: &parallelism,
```
Comment (fejta) [inline test/e2e/batch_v1_jobs.go]: Is this what makes it batch?

File: test/e2e/batch_v1_jobs.go [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+		Expect(err).NotTo(HaveOccurred())
+
+		By("delete a job")
+		reaper, err := kubectl.ReaperFor(extensions.Kind("Job"), f.Client)
+		Expect(err).NotTo(HaveOccurred())
+		timeout := 1 * time.Minute
+		err = reaper.Stop(f.Namespace.Name, job.Name, timeout, api.NewDeleteOptions(0))
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Ensuring job was deleted")
```
Comment (fejta) [inline test/e2e/batch_v1_jobs.go]: Is there a possibility of a race between reaper.Stop(job.Name) and f.Client.Batch().Jobs().Get(job.Name)?

**Comment by fejta** [comment]
@erictune A couple small questions, feel free to add an LGTM tag if you don't want to address them.

File: test/e2e/batch_v1_jobs.go (hunk 1 above)
Comment (erictune) [inline test/e2e/batch_v1_jobs.go]: not necessary.

File: test/e2e/batch_v1_jobs.go (hunk 3 above)
Comment (erictune) [inline test/e2e/batch_v1_jobs.go]: Possible, but I have not ever seen a flake for the identical code in test/e2e/jobs.go.

**Comment by erictune** [comment]
Please cherrypick this so that it becomes part of the test suite for 1.2

**Comment by eparis** [comment]
This PR has added the `cherrypick-candidate` label but did not set a milestone. This is an invalid state and will not be considered for cherrypick-ing to any release branch. The `cherrypick-candidate` label will be automatically removed from such PRs in the future.

**Comment by eparis** [comment]
looks kinda like a new flake to me but at least #21809 #22302 #22226 are closeish
and #22210 #20273 meet the github search term.  I'll let @erictune make decision on how to handle the failure.

**Comment by erictune** [comment]
This PR just adds more e2e tests, so quite unlikely to cause other e2e tests to fail.  So, assuming a flake.

Since several flakes are fixed for deployment, assuming it was one of them.

Rebased and pushed again.

**Pull request merged**